filament/src/components/RenderableManager.cpp

```diff
@@ -458,6 +591,20 @@ void FRenderableManager::create(
             }
         }
 
+        if (UTILS_UNLIKELY(boneCount > 0) && (builder->mBoneIndicesAndWeightsCount > 0)){
+            // create and set texture for bone indices and weights
+            Bones& bones = manager[ci].bones;
+            FSkinningBuffer::HandleIndicesAndWeights handle = downcast(builder->mSkinningBuffer)->
+                createIndicesAndWeightsHandle(downcast(engine), builder->mBoneIndicesAndWeightsCount);
+            bones.handleSamplerGroup = handle.sampler;
+            bones.handleTexture = handle.texture;
+
+            downcast(builder->mSkinningBuffer)->
+                setIndicesAndWeightsData(downcast(engine), handle.texture,
+                       builder->mBoneIndicesAndWeights, builder->mBoneIndicesAndWeightsCount);
+            free(builder->mBoneIndicesAndWeights);
```

Comment:
I used delete[] before, but with calloc() we should() use free right? 
I tried to rewrite it with 
std::unique_ptr<float2[]> mBoneIndicesAndWeights
but I can't call
setIndicesAndWeightsData(downcast(engine), handle.texture,
                       std::move(builder->mBoneIndicesAndWeights), builder->mBoneIndicesAndWeightsCount);
 "copy constructor is implicitly deleted because 'unique_ptr<filament::math::details::TVec2<float>[]>' has a user-declared move constructor unique_ptr(unique_ptr&& __u) _NOEXCEPT"
Any suggestion? Thanks

Reply:
yes thanks, I like that much better.